src/controllers/productsController.ts

```diff
@@ -12,107 +12,125 @@ dotenv.config();
 
 applyFileSysyem();
 
+
 export const getAllProducts = async (req: Request, res: Response) => {
-  let page = Number(req.query.page) || 1;
-  let limit = Number(req.query.limit) || 9;
+  try {
+    let page = Number(req.query.page) || 1;
+    let limit = Number(req.query.limit) || 9;
 
-  if (page < 0) {
-    page = 1;
-  }
-  if (limit > 80 || limit < 0) {
-    limit = 9;
-  }
+    if (page < 0) {
+      page = 1;
+    }
+    if (limit > 80 || limit < 0) {
+      limit = 9;
+    }
 
-  const search = req.query.search?.toString();
-  let conditions: any = {};
+    const search = req.query.search?.toString();
+    let conditions: any = {};
 
-  if (search) {
-    conditions.name = {
-      [Op.like]: `%${req.query.search}%`,
-    };
-  }
+    if (search) {
+      conditions.name = {
+        [Op.like]: `%${req.query.search}%`,
+      };
+    }
 
-  let greaterThan;
-  if (!Number.isNaN(req.query["gte"])) {
-    greaterThan = Number(req.query["gte"]);
-  }
+    let greaterThan;
+    if (!Number.isNaN(req.query['gte'])) {
+      greaterThan = Number(req.query['gte']);
+    }
 
-  let lessThan;
-  if (!Number.isNaN(req.query["lte"])) {
-    lessThan = Number(req.query["lte"]);
-  }
+    let lessThan;
+    if (!Number.isNaN(req.query['lte'])) {
+      lessThan = Number(req.query['lte']);
+    }
 
-  if (greaterThan) {
-    conditions = {
-      ...conditions,
-      price: {
-        [Op.gte]: greaterThan,
-      },
-    };
-  }
+    if (greaterThan) {
+      conditions = {
+        ...conditions,
+        price: {
+          [Op.gte]: greaterThan,
+        },
+      };
+    }
 
-  if (lessThan) {
-    conditions = {
-      ...conditions,
-      price: {
-        ...conditions["price"],
-        [Op.lte]: lessThan,
-      },
-    };
-  }
+    if (lessThan) {
+      conditions = {
+        ...conditions,
+        price: {
+          ...conditions['price'],
+          [Op.lte]: lessThan,
+        },
+      };
+    }
 
-  let sort: any = req.query.sort;
-  if (
-    sort &&
-    (sort == "-name" || sort == "name" || sort == "price" || sort == "-price")
-  ) {
-    let dir;
-    sort.includes("-") ? (dir = "DESC") : (dir = "ASC");
-    sort = [[`${sort.replace("-", "")}`, dir]];
-  } else {
-    sort = [["id", "ASC"]];
-  }
+    let sort: any = req.query.sort;
+    if (
+      sort &&
+      (sort == '-name' || sort == 'name' || sort == 'price' || sort == '-price')
+    ) {
+      let dir;
+      sort.includes('-') ? (dir = 'DESC') : (dir = 'ASC');
+      sort = [[`${sort.replace('-', '')}`, dir]];
+    } else {
+      sort = [['id', 'ASC']];
+    }
+
+    const category = req.query.category as unknown as String;
+
+    if (
+      category &&
+      (category == 'Steel' ||
```

Comment:
it would be better to define this list of categories in constant so you could reuse it later and give it a name so we understand what these categories are

Reply:
We will

Reply:
the categories was not agreed on before we had to add ones for testing, but we did recently and will do a complete implementation for it.